src/commander.c

```diff
@@ -305,11 +305,14 @@ static int commander_process_backup_check(const char *key, const char *filename,
             if (wallet_generate_node(key, seed, &node) == DBB_ERROR) {
                 ret = DBB_ERROR;
             } else {
+                uint8_t is_hidden = wallet_is_hidden();
+                uint8_t negated[] = {1, 0};
+                uint8_t is_main = negated[is_hidden];
```

Comment:
wasn't sure if `!is_hidden` is actually constant time or not on our target platform :see_no_evil: this should be (`wallet_is_hidden()` is `0` or `1`)

Reply:
I'd add this as a comment. Very useful when reading in 6 months.

Reply:
Done